sim_robot_launch.py launches a duplicate sea_surface_estimator

## Problem

Running `ros2 launch marine_simulation simulator_launch.py` produces repeated lifecycle-transition errors for `/ben/sea_surface_estimator`:

```
[ERROR] [launch_ros.utilities.lifecycle_event_manager]: Failed to make transition 'TRANSITION_CONFIGURE' for LifecycleNode '/ben/sea_surface_estimator'
[sea_surface_estimator-8]  [ERROR] Unable to start transition 1 from current state inactive: Transition is not registered., at ./src/rcl_lifecycle.c:355
...
[ERROR] [launch_ros.utilities.lifecycle_event_manager]: Failed to make transition 'TRANSITION_ACTIVATE' for LifecycleNode '/ben/sea_surface_estimator'
[sea_surface_estimator-8]  [ERROR] Unable to start transition 3 from current state active: Transition is not registered.
[sea_surface_estimator-27] [ERROR] Unable to start transition 3 from current state unconfigured: Transition is not registered.
```

The log reveals two `sea_surface_estimator` processes (`-8` and `-27`) both registered as `/ben/sea_surface_estimator`, each running their own `LifecycleTransition(CONFIGURE, ACTIVATE)` and tripping over each other's state.

## Root cause

`sea_surface_estimator` is launched twice under the same fully-qualified name:

1. `marine_simulation/launch/sim_robot_launch.py:349-386` — inline `LifecycleNode` inside a `PushROSNamespace('ben')` group.
2. `ben_project11/launch/ben_core_launch.py:141-149` — includes `mru_transform/sea_surface_estimator_launch.py`, which `sim_robot_launch.py` pulls in via `ben_core_launch`.

The sim block predates the `ben_core_launch` include — ben_project11 added the tide plumbing later (2026-04-15) and the sim-side original was not removed.

The duplicate is only visible in the sim context because ben/bizzyboat/echoboat platforms each launch `sea_surface_estimator` exactly once via their own core launch.

## Fix

Delete the duplicate `GroupAction` block from `marine_simulation/launch/sim_robot_launch.py` (lines 349-386). The `ben_core_launch` include provides the same node with an equivalent `sea_surface_frame = ben/map_tide` parameter, same namespace, same lifecycle transitions.

## Impact

- Spurious "Failed to make transition" / "Transition is not registered" errors at every sim startup.
- Two `sea_surface_estimator` processes running simultaneously, both publishing the same TF and racing each other.

---
**Authored-By**: `Claude Code Agent`
**Model**: `Claude Opus 4.7 (1M context)`

--- a/marine_simulation/launch/sim_robot_launch.py
+++ b/marine_simulation/launch/sim_robot_launch.py
@@ -39,15 +39,12 @@ from launch.substitutions import LaunchConfiguration
 from launch.substitutions import PathJoinSubstitution
 from launch.substitutions import PythonExpression
 from launch.substitutions import TextSubstitution
-from launch_ros.actions import LifecycleNode
-from launch_ros.actions import LifecycleTransition
 from launch_ros.actions import Node
 from launch_ros.actions import PushROSNamespace
 from launch_ros.actions import SetParameter
 from launch_ros.actions import SetParametersFromFile
 from launch_ros.actions import SetRemap
 from launch_ros.substitutions import FindPackageShare
-from lifecycle_msgs.msg import Transition
 
 
 def generate_launch_description():
@@ -345,44 +342,6 @@ def generate_launch_description():
                         ]
                     )
                 ]
-            ),
-            # sea_surface_estimator: estimates tide from odom z for
-            # the map_tide frame used by depth/costmap adjustments
-            GroupAction(
-                actions=[
-                    PushROSNamespace(namespace),
-                    LifecycleNode(
-                        package='mru_transform',
-                        executable='sea_surface_estimator',
-                        name='sea_surface_estimator',
-                        namespace='',
-                        respawn=True,
-                        respawn_delay=2,
-                        emulate_tty=True,
-                        parameters=[{
-                            'sea_surface_frame': PythonExpression(
-                                expression=[
-                                    '"', namespace, '/map_tide"'
-                                ]
-                            ),
-                        }],
-                    ),
-                    LifecycleTransition(
-                        lifecycle_node_names=(
-                            PythonExpression(
-                                expression=[
-                                    '"/',
-                                    namespace,
-                                    '/sea_surface_estimator"',
-                                ],
-                            ),
-                        ),
-                        transition_ids=(
-                            Transition.TRANSITION_CONFIGURE,
-                            Transition.TRANSITION_ACTIVATE,
-                        ),
-                    ),
-                ],
             ),
         ],
         condition=UnlessCondition(no_sim),

